src/sonic_ax_impl/mibs/ietf/rfc4292.py

```diff
@@ -49,26 +50,35 @@ def update_data(self):
             self.route_dest_list.append(sub_id)
             self.route_dest_map[sub_id] = self.loips[loip].packed
 
-        route_entries = Namespace.dbs_keys(self.db_conn, mibs.APPL_DB, "ROUTE_TABLE:*")
-        if not route_entries:
-            return
-
-        for route_entry in route_entries:
-            routestr = route_entry
-            ipnstr = routestr[len("ROUTE_TABLE:"):]
-            if ipnstr == "0.0.0.0/0":
-                ipn = ipaddress.ip_network(ipnstr)
-                ent = Namespace.dbs_get_all(self.db_conn, mibs.APPL_DB, routestr, blocking=True)
-                nexthops = ent["nexthop"]
-                ifnames = ent["ifname"]
-                for nh, ifn in zip(nexthops.split(','), ifnames.split(',')):
-                    ## Ignore non front panel interfaces
-                    ## TODO: non front panel interfaces should not be in APPL_DB at very beginning
-                    ## This is to workaround the bug in current sonic-swss implementation
-                    if ifn == "eth0" or ifn == "lo" or ifn == "docker0": continue
-                    sub_id = ip2tuple_v4(ipn.network_address) + ip2tuple_v4(ipn.netmask) + (self.tos,) + ip2tuple_v4(nh)
-                    self.route_dest_list.append(sub_id)
-                    self.route_dest_map[sub_id] = ipn.network_address.packed
+        # Get list of front end asic namespaces for multi-asic platform.
+        # This list will be empty for single asic platform.
+        front_ns = multi_asic.get_all_namespaces()['front_ns']
+        ipnstr = "0.0.0.0/0"
+        ipn = ipaddress.ip_network(ipnstr)
+        route_str = "ROUTE_TABLE:0.0.0.0/0"
+
+        for db_conn in Namespace.get_non_host_dbs(self.db_conn):
+            # For multi-asic platform, proceed to get routes only for front end namespaces.
+            if front_ns and db_conn.namespace not in front_ns: continue
+            port_config = multi_asic.get_port_table(db_conn.namespace)
```

Comment:
>port_config [](start = 12, length = 11)

Could you filter the port_table with external ports only here?

Reply:
There is another function "multi_asic.is_port_internal()" which will check for external port directly, because if we pass a port channel name to this function, then it will throw an error - "Unknown port name". So before invoking that function we will have to check if the interface name is of a port channel or port name.

Reply:
I don't mean to change the function. Base on the `get_port_table()` result, you can filter based on role.

---
In reply to: [523329081](https://github.com/Azure/sonic-snmpagent/pull/176#discussion_r523329081) [](ancestors = 523329081)

Reply:
Right, but that will add a loop, to loop through all entries and get the role and get the sub-set.
Instead, we can just look for a specific port name once we have a port name and get the sub-role specific to that port.